src/transforms/kubernetes/mod.rs

```diff
@@ -1 +1,552 @@
 pub mod watch_client;
+
+use self::watch_client::{ClientConfig, PodEvent, RuntimeError, Version, WatchClient};
+use super::Transform;
+use crate::{
+    event::{self, Event, Value},
+    sources::kubernetes::POD_UID,
+    topology::config::{DataType, TransformConfig, TransformContext, TransformDescription},
+};
+use bytes::Bytes;
+use evmap::{ReadHandle, WriteHandle};
+use futures::stream::Stream;
+use futures03::compat::Future01CompatExt;
+use k8s_openapi::api::core::v1::Pod;
+use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::collections::VecDeque;
+use std::time::{Duration, Instant};
+use string_cache::DefaultAtom as Atom;
+use tokio::timer::Delay;
+
+// *********************** Defined by Vector **************************** //
+/// Node name `spec.nodeName` of Vector pod passed down with Downward API.
+const NODE_NAME_ENV: &str = "VECTOR_NODE_NAME";
+
+/// If watcher errors, for how long will we wait before trying again.
+const RETRY_TIMEOUT: Duration = Duration::from_secs(2);
+
+/// For how long will we hold on to pod's metadata after the pod has been deleted.
+///
+/// 10min sounds a lot, but a metadata entry averages around 300B and in an extreme
+/// scenario when a pod is deleted every second, we would have ~200kB of probably
+/// not usefull data. Which is acceptable. The benefit of such a long delay
+/// is that we will certainly* have the metadata while we are processing the
+/// remaining logs from the deleted pod.
+const DELETE_DELAY: Duration = Duration::from_secs(10 * 60);
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct KubePodMetadata {
+    #[serde(default = "default_fields")]
+    fields: Vec<String>,
+}
+
+inventory::submit! {
+    TransformDescription::new_without_default::<KubePodMetadata>("kubernetes_pod_metadata")
+}
+
+#[typetag::serde(name = "kubernetes_pod_metadata")]
+impl TransformConfig for KubePodMetadata {
+    fn build(&self, cx: TransformContext) -> crate::Result<Box<dyn Transform>> {
+        // Main idea is to have a background task which will premptively
+        // acquire metadata for all pods on this node, and with it maintaine
+        // a map of extracted metadata.
+
+        // Construct WatchClient
+        let node = std::env::var(NODE_NAME_ENV)
```

Comment:
No, we are setting it [here](https://github.com/timberio/vector/blob/c044571879671bb0416a8cca12a3d3b829682baa/config/kubernetes/vector-daemonset.yaml#L104). 

Config option sounds nice, but I would see it as a feature because, with it, this transform could be used outside Kubernetes cluster which would at least require adding an ability to load kube config file. I think users would expect/assume such ability.

Reply:
I personally think if we are going to force uses to set a env var we should also allow it to be configurable via the toml config too.